packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -41,18 +85,20 @@ class UnpackMacOS extends Target {
   List<Source> get outputs => const <Source>[
     Source.pattern('$_kOutputPrefix/FlutterMacOS'),
     // Headers
+    Source.pattern('$_kOutputPrefix/Headers/FLEDartProject.h'),
+    Source.pattern('$_kOutputPrefix/Headers/FLEEngine.h'),
     Source.pattern('$_kOutputPrefix/Headers/FLEViewController.h'),
     Source.pattern('$_kOutputPrefix/Headers/FlutterBinaryMessenger.h'),
     Source.pattern('$_kOutputPrefix/Headers/FlutterChannels.h'),
     Source.pattern('$_kOutputPrefix/Headers/FlutterCodecs.h'),
-    Source.pattern('$_kOutputPrefix/Headers/FlutterMacOS.h'),
```

Comment:
Sorry, my earlier comment was slightly misleading; this should still be here. This is the umbrella header, and then it lists all the other headers (but not itself, so I shouldn't have said that its contents were the complete list).

Reply:
Done